.github/workflows/build-and-test.yml

```diff
@@ -9,59 +9,48 @@
 name: Build and Test
 
 on:
-  workflow_call:
+  push:
+    branches:
+      - main
```

Comment:
I think this would be redundant with the "main" workflow? Therefore, I think the "main" workflow can be removed.